gfx-descriptor/src/allocator.rs

```diff
@@ -283,6 +283,11 @@ impl<B: Backend> DescriptorAllocator<B> {
             return Ok(());
         }
 
+        assert!(
```

Comment:
We can't panic here. We still need to grow the pools and push the descriptors to the list.

Reply:
I thought trying to allocate a descriptor with an empty layout was a programming error, and we want to panic to help inform the developer. If this is not the case, what should the proper handling be?

Reply:
IIRC, discussion in https://github.com/gpuweb/gpuweb/issues/409 was converging towards allowing this.
Now the question becomes: at what level should this be handled? Technically, could be `gfx`, `gfx-extras`, or `wgpu` itself.
I don't know the answer for sure, it requires some experimentation. Trying to address it here in `gfx-extras` is a part of the equation. It should behave normally as with any other bind group (and layout).